src/core/execute.c

```diff
@@ -2033,6 +2034,16 @@ static int setup_private_users(uid_t uid, gid_t gid) {
         return 0;
 }
 
+static int setup_uts_namespace(const char *hostname) {
+        int r;
+
+        r = unshare(CLONE_NEWUTS);
+        if (r < 0)
+                return r;
```

Comment:
needs to return `-errno` not, `r`, as unshare() returns -1 on error, and the error code in `errno`, but we usually propagate negative error numbers.